internal/aad/aad.go

```diff
@@ -86,8 +88,25 @@ func (auth AAD) Authenticate(ctx context.Context, cfg config.AAD, username, pass
 				logger.Debug(ctx, "Authentication successful even if requiring MFA")
 				return nil
 			}
+			if errcode == noConsentCode {
+				logger.Err(ctx, "Azure AD application requires consent, either from tenant, or from user. "+
+					"If you're a tenant's administrator, go to: %s/adminconsent?client_id=%s",
+					authority, cfg.AppID)
+				return ErrDeny
+			}
+			if errcode == noClientSecretCode {
+				logger.Err(ctx, "Azure AD application requires enabling 'Allow public client flows'. "+
+					"https://learn.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-app-registration#redirect-uris")
+				return ErrDeny
+			}
 		}
 		logger.Err(ctx, "Unknown error code(s) from server: %v", addErrWithCodes.ErrorCodes)
+
+		logger.Debug(ctx, "For more information about the error code(s), see:")
+		for _, errcode := range addErrWithCodes.ErrorCodes {
+			logger.Debug(ctx, "- https://login.microsoftonline.com/error?code=%d", errcode)
```

Comment:
Can you improve the debug log with the error code specified out of the URL as well?
i.e.:
`"Error code: %d  - https://login.microsoftonline.com/error?code=%d"`

Even though this creates a bit of redundancy, it improves the readability and makes it easier for the user to spot the URL that relates to the code that matters.

Reply:
How about the one I updated?

Reply:
Looks good as well, thanks!